### ISSUE TITLE
Error "Invalid authentication" and no login in integration.

### ISSUE THREAD

**Issue by Gferretta** [tuya/tuya-home-assistant#26]
Estimates.
How are you?
I have followed the steps described in the help to configure the new repository in HACS and load the TUYA v2 integration.
Restart HA
Then I went to Configuration -> Integration and looked for Tuya v2

Before this I registered the user in the TUYA android app with username / password
I also registered the user / password in TUYA IOT PLATFORM, creating a new Cloud project, authorizing the 6 APIs mentioned and relating the Link devices by App Tuya Account

The problem starts when I want to configure rest of the integration in HA, following the steps in
1st: Smart Home PaaS
2nd: loading all the requested access data

It always tells me **"Invalid authentication"**

In the "Country Code" field I tried both the name and the international code. Argentina, Argentine, AR, ar, +54, 54
I don't know if it has to do with that.

Obviously the rest of codes are correct.

Another things that happened is logo for Tuya v2 is not correct in Integration menu.

Any idea what may be going on?
Thank you

![Tuya v2 Error](https://user-images.githubusercontent.com/65413059/123372668-60a2ff80-d55a-11eb-8ef2-b82abd9e49c6.jpg)

![Tuya v2 Error 2](https://user-images.githubusercontent.com/65413059/123373470-d491d780-d55b-11eb-8f15-23d4dea56960.jpg)

**Comment by Fannangir** [comment]
I have opened case 21 with exactly the same issue.
Have tried similar stuff, but didn't manage to get it to work :(

**Comment by SgtBatten** [comment]
I  believe that country code should be a number. like the start of an international phone number for you. 1 = USA 86 = china etc

Despite this, i am having the same issue.

**Comment by WaveLicker** [comment]
I also have the same issue.

**Comment by Gferretta** [comment]
Followed this link and video:
https://developer.tuya.com/en/docs/iot/Home_Assistant_Integration?id=Kamcjcbvk2mu8

Put the country code 54 for Argentine.
6 API actived

But still gived me that error and NO connect.

😢

**Comment by Fannangir** [comment]
Did someone manage to connect?
What are we missing?

**Comment by bmorgan99** [comment]
I struggled with this and even opened a ticket on it, but mostly got RTFM.  Changed the Smart Tuya app password, plugged that into the integration setup and voila.

**Comment by Fannangir** [comment]
Unfortunately I have already tried to change the password, no luck :(

**Comment by wiflye81** [comment]
For me it works, just had to recreate a new project and link my tuya account on it. Old project (created before May 25th are not valid). No change on anything else.

**Comment by wiflye81** [comment]
Not the account, just a new project inside the account.

**Comment by Gferretta** [comment]
> I struggled with this and even opened a ticket on it, but mostly got RTFM. Changed the Smart Tuya app password, plugged that into the integration setup and voila.

Just changed my app Tuya password and NO WORK!  :-1: 



> For me it works, just had to recreate a new project and link my tuya account on it. Old project (created before May 25th are not valid). No change on anything else.

What do you mean by recreate?
The actual project can not it deleted.
My account is TRIAL and if i want to create new project, no option of "Development Method: Smart Home PaaS"

How do I delete the current project?


![Tuya v2 Error 3](https://user-images.githubusercontent.com/65413059/123694213-6ab15080-d82f-11eb-986c-b5f79b939717.jpg)

**Comment by Gferretta** [comment]
Tuya original integration work fine inmediately with same user/password. 
But only recognized few devices.
Door Sensors not work.

![SmartSelect_20210628-231056_Home Assistant.jpg](https://user-images.githubusercontent.com/65413059/123726753-de228480-d866-11eb-857b-cf380840acda.jpg)


![SmartSelect_20210628-230727_Home Assistant.jpg](https://user-images.githubusercontent.com/65413059/123726769-e5499280-d866-11eb-87ef-3cf0b77e11f3.jpg)

**Comment by SgtBatten** [comment]
@wiflye81 maybe it would help if you shared more detail such as what country you set everything up as, what region for deployment you selected. 

Did your password contain any special characters or perhaps just letters and numbers.

**Comment by Fannangir** [comment]
In my case, country is Israel.
I can see all the devices under Europe region on the tuya cloud development platform, so I select Europe.
Country code is 972 (Israel).
As for the password, I do have special characters (such as ! and $).
Might there be an issue with special characters?

**Comment by SgtBatten** [comment]
I got mine working by reading the instructions better. Region is predetermined by your country. 

https://github.com/tuya/tuya-home-assistant/issues/10#issuecomment-870269223

**Comment by wiflye81** [comment]
Sorry for the lack of details.
Firstly I follow this video: https://www.youtube.com/watch?v=Amc_fmYMQEo that help me a lot.
My password is alpha numeric with special character, no issue here, and I am from France (Code 33) and choose "Europe" region
My main blocking was to create a new project and recreate a new one.
I also renew my free trial subscription.

**Comment by Gferretta** [comment]
> Sorry for the lack of details.
> Firstly I follow this video: https://www.youtube.com/watch?v=Amc_fmYMQEo that help me a lot.
> My password is alpha numeric with special character, no issue here, and I am from France (Code 33) and choose "Europe" region
> My main blocking was to create a new project and recreate a new one.
> I also renew my free trial subscription.

Well.
After countless tests and counter tests, I was able to get it to successfully login to the Tuya v2 integration.
The first thing I did was unlink the Tuya app account from the Cloud project.
Then he let me create a new project by following exactly the steps in the video and setting the 3 zones.
After that, I linked the Tuya app account to the project.
And there I tried to install the integration and this time EVERYTHING OK.

And the changes of states are immediately.

Thank you very much to all
I think that "something" in the process of creating the project is what prevented the login.

Regards
